Userland/Libraries/LibJS/Runtime/Intl/Segmenter.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2022, Idan Horowitz <idan.horowitz@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/String.h>
+#include <LibJS/Runtime/Object.h>
+
+namespace JS::Intl {
+
+class Segmenter final : public Object {
+    JS_OBJECT(Segmenter, Object);
+
+public:
+    enum class SegmenterGranularity {
+        Grapheme,
+        Word,
+        Sentence,
+    };
+
+    explicit Segmenter(Object& prototype);
+    virtual ~Segmenter() override = default;
+
+    String const& locale() const { return m_locale; }
+    void set_locale(String locale) { m_locale = move(locale); }
```

Comment:
Not really a review comment, but you might find you need to store the `data_locale` from `ResolveLocale` as well. That string is similar, but has Unicode extensions removed, which provides better LibUnicode lookups. It's been important on some Intl objects, so just something to keep in mind if you hit locale specific issues.